### PR TITLE
fix(node/fs): Enable `test-fs-read.js`

### PR DESCRIPTION
**Pull request by wafuwafu13**
part of: https://github.com/denoland/deno_std/issues/911

### CODE REVIEW

File: node/_tools/test/parallel/test-fs-read.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+assert.throws(
+  () => fs.read(fd, Buffer.alloc(1), 0, 1, 0),
+  {
+    code: 'ERR_INVALID_ARG_TYPE',
+  }
+);
+
+// TODO(wafuwafu13): Enable this
+// assert.throws(
```
Comment (wafuwafu13) [inline node/_tools/test/parallel/test-fs-read.js:88]: [memo]
```zsh
Node.js compatibility "parallel/test-fs-read.js" ... FAILED (116ms)
checkConfigTestFilesOrder ... ok (4ms)

 ERRORS 

Node.js compatibility "parallel/test-fs-read.js" => ./node/_tools/test.ts:45:8
error: AssertionError: Failed assertion: error: Uncaught AssertionError: The input did not match the regular expression /TypeError: Cannot read properties of null \(reading 'byteLength'\)/. Input:

'AssertionError [ERR_ASSERTION]: function should not have been called at unknown
called with arguments: null, 4, <Buffer 78 79 7a 0a 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... >'
```

File: node/_tools/test/parallel/test-fs-read.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+//   () => fs.read(fd, { buffer: null }, common.mustNotCall()),
+//   /TypeError: Cannot read properties of null \(reading 'byteLength'\)/,
+//   'throws when options.buffer is null'
+// );
+
+// assert.throws(
```
Comment (wafuwafu13) [inline node/_tools/test/parallel/test-fs-read.js:94]: [memo]
```zsh
Node.js compatibility "parallel/test-fs-read.js" ... FAILED (168ms)
checkConfigTestFilesOrder ... ok (3ms)

 ERRORS 

Node.js compatibility "parallel/test-fs-read.js" => ./node/_tools/test.ts:45:8
error: AssertionError: Failed assertion: error: Uncaught AssertionError: Values are not equal:


    [Diff] Actual / Expected


-   Cannot read properties of undefined (reading 'offset')
+   The"buffer"argument must be an instance of Buffer,TypedArray,or DataView. Received an instance of Object
```

File: node/_tools/test/fixtures/x.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+xyz
```
Comment (wafuwafu13) [inline node/_tools/test/fixtures/x.txt:1]: ref: https://github.com/nodejs/node/blob/main/test/fixtures/x.txt

File: node/_fs/_fs_read.ts [hunk 4]
```diff
@@ -118,7 +127,7 @@ export function read(
     (callback as (err: Error) => void)(err);
   } else {
     const data = Buffer.from(buffer.buffer, offset, length);
-    cb(null, numberOfBytesRead, data);
+    cb(null, numberOfBytesRead ?? 0, data);
```
Comment (wafuwafu13) [inline node/_fs/_fs_read.ts:130]: before:

```js
{
  // Reading beyond file length (3 in this case) should return no data.
  // This is a test for a bug where reads > uint32 would return data
  // from the current position in the file.
  const pos = 0xffffffff + 1; // max-uint32 + 1
  const nRead = fs.readSync(fd, Buffer.alloc(1), 0, 1, pos);

  fs.read(fd, Buffer.alloc(1), 0, 1, pos, common.mustSucceed((nRead) => {
    assert.strictEqual(nRead, 0);
  }));
}
```

```zsh
Node.js compatibility "parallel/test-fs-read.js" ... FAILED (163ms)
checkConfigTestFilesOrder ... ok (4ms)

 ERRORS 

Node.js compatibility "parallel/test-fs-read.js" => ./node/_tools/test.ts:45:8
error: AssertionError: Failed assertion: error: Uncaught AssertionError: Values are not strictly equal:


    [Diff] Actual / Expected


-   null
+   0
```

**Review by kt3k — APPROVED** [review]
LGTM

**Review by cjihrig — APPROVED** [review]

**Pull request merged**